hathor/event/event_manager.py

```diff
@@ -0,0 +1,55 @@
+import time
+
+from structlog import get_logger
+
+from hathor.event.base_event import BaseEvent
+from hathor.event.storage import EventStorage
+from hathor.pubsub import EventArguments, HathorEvents, PubSubManager
+
+logger = get_logger()
+
+
+class EventManager:
+    def __init__(self, event_storage: EventStorage, peer_id: str):
+        self.event_storage = event_storage
+        last_event = event_storage.get_last_event()
+        self.next_event_index = 1 if last_event is None else last_event.id+1
+        self.peer_id = peer_id
+        self.log = logger.new()
+
+    def subscribe(self, pubsub: PubSubManager) -> None:
+        """ Subscribe to defined events for the pubsub received
+        """
+        self.log.error(pubsub)
+        events = [
+            HathorEvents.NETWORK_NEW_TX_ACCEPTED,
+            HathorEvents.NETWORK_NEW_TX_VOIDED,
+            HathorEvents.NETWORK_BEST_BLOCK_FOUND,
+            HathorEvents.NETWORK_ORPHAN_BLOCK_FOUND,
```

Comment:
What if we have a tie? Which event would be emitted?

Reply:
A tie means both blocks are voided, and neither is the best. While it's not too precise to say they are orphans, I think we could go with `NETWORK_ORPHAN_BLOCK_FOUND` and then let a re-org deal with it later.

In practice, I really think we should get rid of ties. There's no benefit in allowing ties to exist, it only adds weird edge cases.

Reply:
We have to discuss how to get rid of it. Let's say we keep the latest best block. In this case, the consensus would depend on the downloading order. Even though it would be rare, it seems like an issue.

Reply:
Yeah, definitely. This should definitely be discussed better, but there are certainly many ways to do it without making it depend on order of operations, like using the numeric hash of the block.

Reply:
That's right! Let's do it somewhere else. Thanks!